lms/views/api/gateway.py

```diff
@@ -58,35 +59,80 @@ def h_lti(context, request):
     # anything until they launch an assignment and get put in a group.
     request.find_service(name="lti_h").sync([context.course], request.lti_params)
 
-    # Add API end-point details
-    h_api_url = request.registry.settings["h_api_url_public"]
     return {
-        "api": {
-            "h": {
-                # These sections are arranged so you can use
-                # `requests.Request.request(**data)` and make the correct request
-                "list_endpoints": {
-                    # List the API end-points
-                    "method": "GET",
-                    "url": h_api_url,
-                    "headers": {"Accept": "application/vnd.hypothesis.v2+json"},
+        "api": {"h": _GatewayService.render_h_connection_info(request)},
+        "data": _GatewayService.render_lti_context(request),
+    }
+
+
+class _GatewayService:
```

Comment:
This is heavily hinting this might want to be a service without actually doing it yet. We can choose later to either:

 * Make it a close collaborator (and maybe pick another name)
 * Make it a full service

In the event we do either having the test split up like this makes it easy and also make the main testing of the view simple.